somexchanger99/erp_utils.py

```diff
@@ -14,6 +14,58 @@ class ErpUtils(object):
     def __init__(self):
         self._client = Client(**settings.ERP_CONF)
 
+    def generate_e101_attachments(self, model, date, process, step, **kwargs):
+
+        date_to = kwargs.get('date_to')
+
+        polissa_obj = self._client.model('giscedata.polissa')
+        partner_obj = self._client.model('res.partner')
+        sw_obj = self._client.model('giscedata.switching')
+        sw_step_obj = self._client.model('giscedata.switching.step')
+        swe101 = self._client.model('giscedata.switching.e1.01')
+        switching_wizard = self._client.model('giscedata.switching.wizard')
+
+        e1s_objects = self._get_objects_with_attachment(
+            model, date, process=process, step=step, date_to=date_to
+        )
+        e101_attachments = []
+
+        for e1 in e1s_objects:
+            if not e1.polissa_ref_id:
```

Comment:
may be it's a little bit academic :neckbeard:, but continues in loops tell me smells. Some time you cannot skip of related conditions in a loop, but in this case, we can write an `if` at the begging of the loop to skip the main funcionality
```python
for e1 in e1_objects:
    pass_ids = swe101.search([('sw_id', '=', e1.id)])
    if e1.polissa_ref_id and pass_ids:
        pas_id = pas_ids[0]
        e101_step = sorted(steps)[0]
        step_id = str( (e101_step, pas_id) )
￼.....
```

Reply:
And then here you can refactor this code into a function that has the responsibility of extract an xml of an atr case :smile:

Reply:
Fixed